code/modules/mob/living/simple_animal/hostile/roguetown/jitterskull.dm

```diff
@@ -0,0 +1,1338 @@
+                if(dd >= 12 && world.time >= next_tether_allowed)
+                    next_tether_allowed = world.time + 20
+                    // Snap tether near target without entering a stalking pause
+                    teleport_near_atom(T, 8, 12)
+                    var/df2 = get_dir(src, T)
+                    if(df2)
+                        dir = df2
+        sleep(10)
+
+// Loop that triggers idle chatter at random intervals to build tension
+/mob/living/simple_animal/hostile/rogue/jitterskull/proc/idle_chatter_loop()
+    while(src && !QDELETED(src))
+        if(stat != DEAD && world.time >= next_idle_chatter_time)
+            if(prob(60)) // not every window we chatter
+                var/chatter = pick('sound/mobs/jitter_chatter1.ogg','sound/mobs/jitter_chatter2.ogg','sound/mobs/jitter_chatter3.ogg','sound/mobs/jitter_chatter4.ogg')
+                playsound(src, chatter, 50, FALSE)
+            // schedule next opportunity 3–8 seconds from now
+            next_idle_chatter_time = world.time + rand(30, 80)
+        sleep(rand(20, 40)) // check again in 2–4 seconds
+
+// Periodic leash separate from anti-stuck, to gently keep the skull near its current target when they flee far away
+/mob/living/simple_animal/hostile/rogue/jitterskull/proc/pursuit_tether_loop()
+    while(src && !QDELETED(src))
+        if(stat != DEAD && !is_dying && !is_feasting && !is_guarding && !is_stalking && ai_controller)
+            var/datum/ai_controller/C = ai_controller
+            var/mob/living/T = C.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+            if(T && !QDELETED(T) && world.time >= next_tether_allowed)
+                var/d = get_dist(src, T)
+                // Engage tether at a closer gap so we keep pressure without dropping aggro
+                if(d >= 16)
+                    next_tether_allowed = world.time + 25
+                    // Snap tether without entering stalking
+                    teleport_near_atom(T, 10, 14)
+                    var/df = get_dir(src, T)
+                    if(df)
+                        dir = df
+        sleep(12)
+
+// Custom death sequence: shake, flames, death sound, then burst into gibs
+/mob/living/simple_animal/hostile/rogue/jitterskull/death(gibbed)
+    if(is_dying)
+        return
+    is_dying = TRUE
+    // Cache turf for loot drop before we potentially delete ourselves
+    var/turf/death_turf = get_turf(src)
+    // Halt AI to avoid any further actions during death
+    toggle_ai(AI_OFF)
+    // Ensure we won't move/teleport during death
+    SetImmobilized(0)
+    Immobilize(50)
+    // Spectral flames in cardinal directions (visual only)
+    spawn_flames()
+    // Play death sound
+    playsound(src, 'sound/mobs/jitter_death.ogg', 80, FALSE)
+    // Shake violently for ~2.5 seconds while the audio plays
+    shake_violently(25)
+    // Small spectral burst and gibs
+    visible_message(span_warning("[src] bursts apart in a spectral blast!"))
+    gib_animation()
+    gib()
```

Comment:
`gib()` will call `death()` if `stat` hasn't been set to `DEAD` yet, and because `..()` is never called in this proc, `stat` is never set to `DEAD` when this mob dies. that's probably why the `is_dying` check is there but it would be much better served by just doing `. = ..()` at the start of the proc, i think.